### PR TITLE
Add Jest for API testing and test for Sign Up page

### LINKED ISSUE #104: Jest for API testing

### PR DESCRIPTION
**Pull request by ketphan02**
+ Add a success test for sign up function.
+ Discuss about library to use to fetch api from backend
+ Close #104

### CODE REVIEW

File: src/__tests__/pages/api/auth/signup.test.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+import type { UserDataProp } from '../../../../__mocks__/pages/api/auth';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const prisma = new PrismaClient();
+
+beforeAll(async () => {
+  await prisma.$connect();
+});
+
+beforeEach(async () => {
```
Comment (Opeyem1a) [inline src/__tests__/pages/api/auth/signup.test.ts:22]: Does this run before each `describe()` or before each `it()`?

File: src/__tests__/pages/api/auth/signup.test.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  await prisma.user.deleteMany();
+});
+
+afterAll(async () => {
+  await prisma.$disconnect();
+});
+
+describe('Successfully create user(s)', () => {
+  it('Create user successs', async () => {
```
Comment (Opeyem1a) [inline src/__tests__/pages/api/auth/signup.test.ts:31]: these strings really should read like sentences - "it create user success" is a weird sentence. For example, "it successfully creates a user" is a better sentence, which becomes `it('successfully creates a user', ...)`
This comment applies to every PR/every test

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Comment by Opeyem1a** [comment]
Also resolve the merge conflicts

File: src/__tests__/pages/api/auth/signup.test.ts (hunk 1 above)
Comment (ketphan02) [reply src/__tests__/pages/api/auth/signup.test.ts:22]: before all it

File: src/__tests__/pages/api/auth/reset-password.test.ts [hunk 3]
```diff
@@ -239,7 +239,7 @@ describe('Fail to change password', () => {
     ]);
   });
 
-  it('New password is null or undefined', async () => {
+  it('has null or undefined new password', async () => {
```
Comment (Opeyem1a) [inline src/__tests__/pages/api/auth/reset-password.test.ts:242]: `it('returns an error if new password is null or undefined')`

Comment (Opeyem1a) [reply src/__tests__/pages/api/auth/reset-password.test.ts:242]: This applies to all

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**